README-QUIC.md

```diff
@@ -0,0 +1,98 @@
+  blocking and allows an application to open additional logical streams without
+  any round trip penalty, unlike opening an additional TCP connection.
+
+- Since QUIC is the basis of HTTP/3, support for QUIC also enables applications
+  to use HTTP/3 using a suitable third-party library.
+
+- Future versions of OpenSSL will offer support for 0-RTT connection
+  initiation, allowing a connection to be initiated to a server and application
+  data to be transmitted without any waiting time. This is similar to TLS 1.3's
+  0-RTT functionality but also avoids the round trip needed to open a TCP
+  socket; thus, it is similar to a combination of TLS 1.3 0-RTT and TCP Fast
+  Open.
+
+- Future versions of OpenSSL will offer support for connection
+  migration, allowing connections to seamlessly survive IP address changes.
+
+- Future versions of OpenSSL will offer support for the QUIC
+  datagram extension, allowing support for both TLS and DTLS-style use cases on
+  a single connection.
+
+- Because most QUIC implementations, including OpenSSL's implementation, are
+  implemented as an application library rather than by an operating system, an
+  application can gain the benefit of QUIC without needing to wait for an OS
+  update to be deployed. Future evolutions and enhancements to the QUIC protocol
+  can be delivered as quickly as an application can be updated without
+  dependency on an OS update cadence.
+
+- Because QUIC is UDP-based, it is possible to multiplex a QUIC connection
+  on the same UDP socket as some other UDP-based protocols, such as RTP.
+
+For more background information on OpenSSL's QUIC implementation, see the
+[openssl-quic(7) manpage].
+
+### How can I use HTTP/3 with OpenSSL?
+
+There are many HTTP/3 implementations in C available. The use of one such HTTP/3
+library with OpenSSL QUIC is demonstrated via the [demo found in `demos/http3`].
+
+### How can I use OpenSSL QUIC in my own application for a different protocol?
+
+The [OpenSSL Guide] provides introductory examples for how to make use of
+OpenSSL QUIC.
+
+The [openssl-quic(7) manpage] and the [DDD demos] may also be helpful to
+illustrate the changes needed if you are trying to adapt an existing
+application.
+
+### How can I test QUIC using `openssl s_client`?
+
+There is basic support for single-stream QUIC using `openssl s_client`:
+
+```shell
+$ openssl s_client -quic -alpn ossltest -connect www.example.com:12345
+```
+
+This connects to a QUIC server using the specified ALPN protocol name and opens
+a single bidirectional stream. Data can be passed via stdin/stdout as usual.
+This allows test usage of QUIC using simple TCP/TLS-like usage.
+
+[openssl-quic(7) manpage]: https://www.openssl.org/docs/manmaster/man7/openssl-quic.html
+[OpenSSL guide]: https://www.openssl.org/docs/manmaster/man7/ossl-guide-introduction.html
+[ossl-guide-introduction(7) manpage]: https://www.openssl.org/docs/manmaster/man7/ossl-guide-introduction.html
```

Comment:
this link is a duplicate of the line above.

it should point to a manpage.

Reply:
sorry, maybe this is intentional.

In the second bullet (line 13), there are two links to the same thing and that confused me.

Reply:
Yeah, it's intentional.